api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentStatusController.java

```diff
@@ -0,0 +1,159 @@
+package uk.gov.hmcts.payment.api.controllers;
+
+import io.swagger.annotations.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.hmcts.payment.api.configuration.LaunchDarklyFeatureToggler;
+import uk.gov.hmcts.payment.api.dto.*;
+import uk.gov.hmcts.payment.api.dto.mapper.PaymentStatusResponseMapper;
+import uk.gov.hmcts.payment.api.exception.FailureReferenceNotFoundException;
+import uk.gov.hmcts.payment.api.model.Payment;
+import uk.gov.hmcts.payment.api.model.Payment2Repository;
+import uk.gov.hmcts.payment.api.model.PaymentFailures;
+import uk.gov.hmcts.payment.api.service.PaymentStatusUpdateService;
+import uk.gov.hmcts.payment.api.v1.model.exceptions.PaymentNotFoundException;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@RestController
+@Api(tags = {"Payment Status"})
+@SwaggerDefinition(tags = {@Tag(name = "PaymentStatusController", description = "Payment Status REST API")})
+public class PaymentStatusController {
+    private static final Logger LOG = LoggerFactory.getLogger(PaymentStatusController.class);
+
+    @Autowired
+    private PaymentStatusUpdateService paymentStatusUpdateService;
+
+    @Autowired
+    private Payment2Repository paymentRepository;
```

Comment:
The repository bean should be autowired in ServiceImpl only.

Reply:
implemented